added process steps to contribution (#4)
tesych/Azure-Kinect-Sensor-SDK: CONTRIBUTING.md

### Requesting new features

Please review the list of open Issues to see if one is already open. Please review all categories, Bugs and enhancements. Also check for Closed Issues before opening a new one.
If you have a small enhancement that is well defined, please create a new feature request on GitHub.

> If you have a larger idea for the new feature please share it with us on [Microsoft Azure Feedback Forums](https://feedback.azure.com/forums/920053) where the rest of the community can vote for it as well. We will review the ideas submitted and update the status as we go through the refinement and development process:

Reviewer: ```suggestion
If you have a larger idea for the new feature please share it with us on [Microsoft Azure Feedback Forums](https://feedback.azure.com/forums/920053) where the rest of the community can up vote it too. We will review the submitted ideas at least once a week, usually twice weekly. 

Below is our list of possible status states we may assign to the request:
```

| Microsoft Azure Feedback Forums status| Process steps definition
|---------------------------------------|---------------------------------------|
| Need Feedback                         | Idea- doesn't have enough details or needs community support |
| Under Review                          | Proposal for the new feature is created and in review |
| Planned                               | Approved to get the work started  |
| Started                               | Work started                      |
| Completed                             | Feature is released               |

Once an idea is well defined and reached the popularity vote, a written proposal( is coming soon...) for implementation will be created on GitHub for all members of the community to review and provide feedback on the specific implementation to ensure it covers their needs. Once the the implementation of the feature is approved by both community and Microsoft team, the implementation will start.

> We will consider all of the ideas, but some we will be declined to implement due to the restrictions of the product.

Reviewer: ```suggestion
```

If you have a small enhancement that is well defined, please create a new feature request on GitHub.

If you have a larger idea for the new feature please share it with us on [Microsoft Azure Feedback Forums](https://feedback.azure.com/forums/920053) where the rest of the community can up vote it too. We will review the submitted ideas at least once a week, usually twice weekly. 

Below is our list of possible status states we may assign to the request:

| Microsoft Azure Feedback Forums status| Process steps definition
|---------------------------------------|---------------------------------------|
| <blank>                                    | New issue we have not looked at yet |
| Need Feedback                         | Idea- doesn't have enough details or needs community support |
| Under Review                          | Proposal for the new feature is created and in review |
| Planned                               | Approved to get the work started  |
| Started                               | Work started                      |
| Completed                             | Feature is released               |

> | Declined                                 | We have decided not to make this change |

Reviewer 2: | Moved                                            | Issue which has been moved to GitHub |

> 

Reviewer 2: | Archived                                         | Feature not possible to implement on current HW. | 
| Triaged                                           | Request has been seen and is under internal discussion. |

### Requesting new features

Please review the list of open Issues to see if one is already open. Please review all categories, Bugs and enhancements. Also check for Closed Issues before opening a new one.
If you have a small enhancement that is well defined, please create a new feature request on GitHub.

> If you have a larger idea for the new feature please share it with us on [Microsoft Azure Feedback Forums](https://feedback.azure.com/forums/920053) where the rest of the community can up vote it too. We will review the submitted ideas very frequently, usually twice weekly. 

Reviewer: ```suggestion
If you have a larger idea for a new feature please share it with us on [Microsoft Azure Feedback Forums](https://feedback.azure.com/forums/920053) where the rest of the community can up vote it too. We will review the submitted ideas very frequently, usually twice weekly. 
```

Below is our list of possible status states we may assign to the request:

| Microsoft Azure Feedback Forums status| Process steps definition
|---------------------------------------|---------------------------------------|
| No status                                 | New issue we have not looked at yet |

> | Need Feedback                         | Idea doesn't have enough details or needs community support |

Reviewer 3: ```suggestion
| Need Feedback                         | Idea doesn't have enough details or needs more community support |
```

> | Need Feedback                         | Idea- doesn't have enough details or needs community support |

Reviewer: ```suggestion
| Need Feedback                         | Idea doesn't have enough details or needs community support |
```

| Under Review                          | Proposal for the new feature is created and in review |
| Planned                               | Approved to get the work started  |
| Started                               | Work started                      |
| Completed                             | Feature is released               |
| Declined                              | We have decided not to make this change |
| Moved                                 | Issue which has been moved to GitHub |
| Archived                              | Feature not possible to implement on current HW. |
| Triaged                               | Request has been seen and is under internal discussion |

#### No status

This is a new request that we have not yet seen. Within a week we should have reviewed the request and assigned it an initial state.

#### Needs Feedback

> The Azure Kinect team uses this state to ask for more information about this issue. We may need more information from the issue filer as we don't completely understand the request. We might also use this status because we are waiting for more community feedback on the proposal; either in the form or spec feedback or up voting.

Reviewer 3: ```suggestion
The Azure Kinect team uses this state to ask for more information about this issue. We may need more information from the issue filer as we don't completely understand the request. We might also use this status because we are waiting for more community feedback on the proposal; either in the form of spec feedback or up voting.
```

#### Under Review

> The team is actively reviewing the proposal and determine what the next actions should be. We may ask for more information, iterate on the proposal, or move to planned state while we wait for resources. We will also need to carefully consider not only the work to implement the request but the work needed to be invested our build and test infrastructure to ensure quality of the code remains high.

Reviewer: When the first proposal goes live, I would like all instances of 'proposal' to be a link to the proposal folder.

#### Planned

> The quested issue has been planned but not yet started. We may stay in this state indefinitely if we don't have resources to complete the request.

Reviewer 3: ```suggestion
The requested issue has been planned but not yet started. It may stay in this state indefinitely if we don't have resources to complete the request.
```

> The quested issue have been planned but not yet started. We may stay in this state indefinitely if we don't have resources to complete the request.

Reviewer: ```suggestion
The quested issue has been planned but not yet started. We may stay in this state indefinitely if we don't have resources to complete the request.
```

#### Started

Work on the feature, new tests, and possible infrastructure changes have begun.

#### Completed

The requested issue has been checked into develop branch and we are done. The feature however, may not yet be part of a release.

#### Declined

> We have consider the request and have decided not to implement it due to various reasons.

Reviewer: I think we should list what some of these reasons might be. As an example the testing required for a simple feature might be too high, and as a result we can't justify the cost.

Author: will add some examples

#### Moved

Issue has been moved to GitHub for tracking.

#### Archived

> We will archive the idea if it doesn't fit for the current hardware.

Reviewer: ```suggestion
We will archive the idea if the current hardware can't support the request.
```
